src/esp/core/Configuration.h

```diff
@@ -9,184 +9,619 @@
+    for (const auto& entry : valueMap_) {
+      keys.push_back(entry.first);
     }
-    return configPtr;
+    return keys;
+  }
+  std::vector<std::string> getBoolKeys() const {
+    return getKeysFromMapOfType(ConfigStoredType::BOOL);
+  }
+  std::vector<std::string> getDoubleKeys() const {
+    return getKeysFromMapOfType(ConfigStoredType::DOUBLE);
+  }
+  std::vector<std::string> getIntKeys() const {
+    return getKeysFromMapOfType(ConfigStoredType::INT);
+  }
+  std::vector<std::string> getStringKeys() const {
+    return getKeysFromMapOfType(ConfigStoredType::STRING);
+  }
+  std::vector<std::string> getVec3Keys() const {
+    return getKeysFromMapOfType(ConfigStoredType::MN_VEC3);
   }
-  void setConfigSubSubgroup(Configuration& config,
-                            const std::string& subGroupName,
-                            const std::string& subSubGroupName) {
-    cfg.group(subGroupName)->addGroup(subSubGroupName);
-    *cfg.group(subGroupName)->group(subSubGroupName) =
-        *config.cfg.group(subSubGroupName);
+  std::vector<std::string> getQuatKeys() const {
+    return getKeysFromMapOfType(ConfigStoredType::MN_QUAT);
+  }
+  std::vector<std::string> getRadKeys() const {
+    return getKeysFromMapOfType(ConfigStoredType::MN_RAD);
+  }
+  // ****************** Setters ******************
+  template <typename T>
+  void set(CORRADE_UNUSED const std::string& key,
+           CORRADE_UNUSED const T& value) {
+    ESP_ERROR() << "Unknown/unsupported type :" << typeid(T).name()
+                << "for key :" << key;
+  }
+  void set(const std::string& key, const bool& value) {
+    valueMap_[key].set(value);
+  }
+  void set(const std::string& key, const int& value) {
+    valueMap_[key].set(value);
+  }
+
+  void set(const std::string& key, const double& value) {
+    valueMap_[key].set(value);
+  }
+  void set(const std::string& key, const char* value) {
+    valueMap_[key].set(value);
+  }
+
+  void set(const std::string& key, const std::string& value) {
+    valueMap_[key].set(value);
   }
 
-  int getNumConfigSubgroups(const std::string& name) const {
-    if (cfg.hasGroup(name)) {
-      return cfg.group(name)->valueCount();
+  void set(const std::string& key, const Magnum::Vector3& value) {
+    valueMap_[key].set(value);
+  }
+  void set(const std::string& key, const Magnum::Quaternion& value) {
+    valueMap_[key].set(value);
+  }
+  void set(const std::string& key, const Magnum::Rad& value) {
+    valueMap_[key].set(value);
+  }
+  // ****************** Value removal ******************
+  void removeValueFromMap(const std::string& key) {
+    if (valueMap_.count(key) > 0) {
+      valueMap_.erase(key);
     }
-    return 0;
   }
 
-  /**@brief Add a string to a group and return the resulting group size. */
-  int addStringToGroup(const std::string& key, const std::string& value) {
-    cfg.addValue(key, value);
-    return cfg.valueCount(key);
+  bool removeBool(const std::string& key) {
```

Comment:
Why do we need these type-specific removes?

Reply:
wanted to provide a remove w/return object removed.